org.eclipse.scanning.api/src/org/eclipse/scanning/api/event/core/ResponseConfiguration.java

```diff
@@ -27,32 +27,34 @@ public boolean waitAgain() {
 			}
 		}
 	}
-	
+
 	public static final ResponseConfiguration DEFAULT = new ResponseConfiguration(ResponseType.ONE, 100, TimeUnit.MILLISECONDS) {
+		@Override
 		public void setTimeout(long timeout) {
 			throw new RuntimeException("Timeout is immutable!");
 		}
+		@Override
 		public void setTimeUnit(TimeUnit timeUnit) {
 			throw new RuntimeException("TimeUnit is immutable!");

```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Define and throw a dedicated exception instead of using a generic one. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS00112)